src/types/redis_search_encoding.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <encoding.h>
+#include <storage/redis_metadata.h>
+
+namespace redis {
+
+inline constexpr auto kErrorInsufficientLength = "insufficient length while decoding metadata";
+
+enum class SearchSubkeyType : uint8_t {
```

Comment:
What do `TAG_FIELD_META` and `TAG_FIELD` means?

Reply:
`TAG_FIELD_META` means metadata of this index (field, typed `tag`), and `TAG_FIELD` simply stores the index.

e.g. for a tag field `x` in key `k`, if we insert json `a = {"x": "hello;hi"}`, then the low-level ops are:
```
write:  k | TAG_FIELD | x | hello | a -> (nil)
write:  k | TAG_FIELD | x | hi | a -> (nil)
```

Reply:
🤔So this is different type of subkey?